lib/bb/server/process.py

```diff
@@ -392,10 +392,10 @@ def __init__(self, connection, recv):
 
     def runCommand(self, command):
         self.connection.send(command)
-        if not self.recv.poll(30):
+        if not self.recv.poll(300):
             logger.info("No reply from server in 30s")
-            if not self.recv.poll(30):
-                raise ProcessTimeout("Timeout while waiting for a reply from the bitbake server (60s)")
+            if not self.recv.poll(300):
+                raise ProcessTimeout("Timeout while waiting for a reply from the bitbake server (300s)")
```

Comment:
The final wait would amount to 600s here, right? How often would the error reproduce in your runs before this change (out of 10)?
I'm also worried that this change is not upstreamable as is since we are unable to determine what causes the timeouts. Why is this wait time not working for us alone?

Reply:
There is only going to be that much wait if the process is really taking that long. In normal builds this wouldn't change anything. Azdo's statistics say 50% of builds failed.
Yes, this likely won't be upstreamable. I suspect it has to do with the increased load on servers.

Reply:
Ok. I like Alex's suggestion below. Btw earlier, I was referring to the timeout number in line 398, which should be changed to 600s.

Reply:
Changed to 600s